src/lib/transition_frontier/persistent_frontier/worker.ml

```diff
@@ -24,23 +27,24 @@ module Worker = struct
    them immediately. This invariant is potentially violated whenever we touch the
    database (i.e., whenever we process any other kind of diff) so we eagerly perform
    work in this queue immediately after applying any other kind of diff. *)
-    ; root_transitions: Diff.Root_transition.Lite.t Queue.t }
+    ; root_transitions: Diff.Root_transition.Lite.t Queue.t
+    ; persistent_root_instance: Persistent_root.Instance.t }
 
   type nonrec create_args = create_args
 
   type output = unit
 
   (* worker assumes database has already been checked and initialized *)
-  let create ({db; logger} : create_args) : t =
-    {db; logger; root_transitions= Queue.create ()}
+  let create ({db; logger; persistent_root_instance} : create_args) : t =
+    {db; logger; root_transitions= Queue.create (); persistent_root_instance}
 
   let eagerly_perform_root_transitions t =
```

Comment:
should this be removed now that root snarked ledger is in sync with the persisted frontier root(preferably in another PR)? eagerly applying root diffs can cause the less-than-k breadcrumbs problem https://github.com/MinaProtocol/mina/issues/6830.

Reply:
Yes, thank you for catching this. We now want to remove the eager root applications in favor of this approach. It's possible simply reverting the other PR would get us in the right state (if there aren't too many conflicts).